system/HTTP/CURLRequest.php

```diff
@@ -385,6 +385,12 @@ public function send(string $method, string $url)
         // Set the string we want to break our response from
         $breakString = "\r\n\r\n";
 
+        if (isset($this->config['allow_redirects']) && $this->config['allow_redirects'] !== false) {
+            while (preg_match('/^HTTP\/\d(?:\.\d)? 3\d\d/', $output)) {
```

Comment:
I am not a fan of this. It seems too broad. “304 Not Modified” should not be removed. IMO the previous implementation was better.

Reply:
Agree that this is broad, that's why I have added checks for `Location` header, which is the behaviour of curl: Any 3xx code with `Location` header will redirect.

As we can't control how target server respond, matching curl behaviour would best in my opinion. I'm happy to change back to the previous if you still think it was better.

Reply:
Ok, since we are now checking `Location`, we can move on to broad checking which will allow us to support edge cases where a server uses a non-standard or uncommon 3xx status code.

Can I ask you to move the whole `while` loop to a separate private method such as `handleRedirectHeaders()`? There are too many things going on here right now.

Reply:
Sweet! I've just pushed the commit moving to a function